### ISSUE TITLE
[Template Instantiation] Review the cloning step

### ISSUE THREAD

**Issue by zaclummys** [WICG/webcomponents#740]
One of the steps in the [Template Instantiation](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md) is to clone **HTMLTemplateElement#content** using **Node#cloneNode**. But in this way, the result won't be upgraded until it's attached to the document. 

On the other hand, if cloning is done using **Document#ImportNode**, the result is cloned and upgraded appropriately.

This can become undesirable, for example, by using custom elements. With the current method, some lifecycles and initialization methods may be deferred.

To be taken as an example,
https://jsbin.com/pobavohufu/edit?js,console

**Comment by rniwa** [comment]
Presumably, you're referring to step 1 in https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md#43-creating-template-parts ? That's different from `cloneNode`. That's the concept [`importNode` uses](https://dom.spec.whatwg.org/#dom-document-importnode). I think what's missing here is that I'm not specifying with respect to which document nodes are cloned. That needs to be specified.

**Comment by zaclummys** [comment]
I assumed [ Creating Template Parts: #1](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md#43-creating-template-parts) uses **Node#cloneNode**, although it isn't explicitly described.

Using **Document#importNode** is a good way to work around this problem?

**Comment by rniwa** [comment]
No, [cloning](https://dom.spec.whatwg.org/#concept-node-clone) is an [abstract concept](https://dom.spec.whatwg.org/#concept-node-clone) used by both `cloneNode` and `importNode`.
